module/modmanager/manager_test.go

```diff
@@ -567,6 +572,53 @@ func TestModuleReloading(t *testing.T) {
 		err = mgr.Close(ctx)
 		test.That(t, err, test.ShouldBeNil)
 	})
+
+	t.Run("cancelled module process is stopped", func(t *testing.T) {
+		logger, logs := logging.NewObservedTestLogger(t)
+
+		modCfg.ExePath = rutils.ResolveFile("module/testmodule/fakemodule.sh")
+
+		// Lower global timeout early to avoid race with actual restart code.
+		defer func(oriOrigVal time.Duration) {
+			oueRestartInterval = oriOrigVal
+		}(oueRestartInterval)
+		oueRestartInterval = 10 * time.Millisecond
+
+		// Lower module startup timeout to avoid waiting for 5 mins.
+		defer func() {
+			test.That(t, os.Unsetenv(rutils.ModuleStartupTimeoutEnvVar),
```

Comment:
FWIW @maximpertsov brought up a better solution for stuff like this a while ago. I forget what test we use it in, but `t.Setenv` will automatically restore the variable after the test.

Reply:
https://github.com/viamrobotics/rdk/pull/3187/commits/e5037eca6ae6b9f00527d0ab797c5f8e9be8fe27

Reply:
thanks! updated